### PR TITLE
Revert "[sdk/dotnet] Add PluginDownloadURL"

### PR DESCRIPTION
**Pull request by t0yv0**
Reverts pulumi/pulumi#8702 temporarily as we sort out why it broke master builds on Mac OS that were excluded from PR verification to economize on Mac workers unfortunately.

### CODE REVIEW

**Comment by t0yv0** [comment]
I'm suspecting something is funny with `+			[]string{dotNetBin, "add", "package", dep, "-v", version, "-s", localNuget}, cwd)` but no proof yet.. The tests seem to work fine locally for me on a Mac, it's confusing.

**Review by justinvp — APPROVED** [review]
🤞

File: pkg/testing/integration/program.go [hunk 1]
```diff
@@ -2075,7 +2075,7 @@ func (pt *ProgramTester) prepareDotNetProject(projinfo *engine.Projinfo) error {
 		version := r.Replace(file)
 
 		err = pt.runCommand("dotnet-add-package",
-			[]string{dotNetBin, "add", "package", dep, "-v", version, "-s", localNuget}, cwd)
```
Comment (iwahbe) [inline pkg/testing/integration/program.go:2078]: I agree that this is probably the problem. It works locally on my Mac, so my initial suspicion is that `localNuget` points to the wrong place on the Mac test suite. But we use `localNuget` to get the version, so it must be correct. 

---

Either way, we can revert to unblock the release and then fix the test.

**Review by iwahbe — APPROVED** [review]

File: pkg/testing/integration/program.go (hunk 1 above)
Comment (t0yv0) [reply pkg/testing/integration/program.go:2078]: I'm sorry about this - there's a drift between local dev env and CI env and my recent work makes it a bit worse even to hit some other objectives. Thank you for your patience.

**Pull request merged**